runtime/v2/manager.go

```diff
@@ -140,14 +141,73 @@ func (m *MM) ValidateGenesis(genesisData map[string]json.RawMessage) error {
 	return nil
 }
 
-// InitGenesis performs init genesis functionality for modules.
-func (m *MM) InitGenesis() {
-	panic("implement me")
+// ExportGenesis performs export genesis functionality for modules
+func (m *MM) ExportGenesis(ctx context.Context) (map[string]json.RawMessage, error) {
+	return m.ExportGenesisForModules(ctx, []string{})
 }
 
-// ExportGenesis performs export genesis functionality for modules
-func (m *MM) ExportGenesis() {
-	panic("implement me")
+// ExportGenesisForModules performs export genesis functionality for modules
+func (m *MM) ExportGenesisForModules(ctx context.Context, modulesToExport []string) (map[string]json.RawMessage, error) {
+	if len(modulesToExport) == 0 {
+		modulesToExport = m.config.ExportGenesis
+	}
+	// verify modules exists in app, so that we don't panic in the middle of an export
+	if err := m.checkModulesExists(modulesToExport); err != nil {
+		return nil, err
+	}
+
+	type genesisResult struct {
+		bz  json.RawMessage
+		err error
+	}
+
+	channels := make(map[string]chan genesisResult)
+	for _, moduleName := range modulesToExport {
+		mod := m.modules[moduleName]
+		if module, ok := mod.(appmodulev2.HasGenesis); ok {
```

Comment:
nit: Looks like `appmodulev2.HasGenesis` and `sdkmodule.HasABCIGenesis` both impl the same `ExportGenesis(context.Context) (json.RawMessage, error)` method.  You could get rid of one if block if you just check for this method (in a new interface?)

Reply:
very make sense, ty

Reply:
resolved